# HG changeset patch
# User Rik <rik@octave.org>
# Date 1501266638 25200
#      Fri Jul 28 11:30:38 2017 -0700
# Node ID bf061a189af9fe6426b9b47cda4638cd018528a4
# Parent  8057d3f0673de352f2431898012d3b122fd0ec26
Use C++11 raw string literals to avoid escaping double quotes.

* input.cc, jit-ir.h:
Replace regular strings containing "\"" patterns with R"(..."...)".

diff --git a/libinterp/corefcn/input.cc b/libinterp/corefcn/input.cc
--- a/libinterp/corefcn/input.cc
+++ b/libinterp/corefcn/input.cc
@@ -524,17 +524,17 @@ initialize_command_input (void)
 
   octave::command_editor::set_basic_word_break_characters (s);
 
   octave::command_editor::set_completer_word_break_characters (s);
 
   octave::command_editor::set_basic_quote_characters (R"(")");
 
   octave::command_editor::set_filename_quote_characters (" \t\n\\\"'@<>=;|&()#$`?*[!:{");
-  octave::command_editor::set_completer_quote_characters ("'\"");
+  octave::command_editor::set_completer_quote_characters (R"('")");
 
   octave::command_editor::set_completion_function (generate_completion);
 
   octave::command_editor::set_quoting_function (quoting_filename);
 }
 
 static void
 execute_in_debugger_handler (const std::pair<std::string, int>& arg)
diff --git a/libinterp/parse-tree/jit-ir.h b/libinterp/parse-tree/jit-ir.h
--- a/libinterp/parse-tree/jit-ir.h
+++ b/libinterp/parse-tree/jit-ir.h
@@ -531,20 +531,20 @@ public:
 
   PASS_T value (void) const { return mvalue; }
 
   virtual std::ostream& print (std::ostream& os, size_t indent = 0) const
   {
     print_indent (os, indent);
     jit_print (os, type ()) << ": ";
     if (QUOTE)
-      os << "\"";
+      os << R"(")";
     os << mvalue;
     if (QUOTE)
-      os << "\"";
+      os << R"(")";
     return os;
   }
 
   JIT_VALUE_ACCEPT;
 private:
   T mvalue;
 };
 
